### PR TITLE
Changing high-contrast mode colors to system colors

### PR DESCRIPTION
**Pull request by kristoferbrown**
#### Pull request checklist
- [x] Fixes issues: #1804, #1829, #2541, #2474, #2115
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Previously, most of our high contrast mode media queries assigned specific colors. This PR changes them to use system colors, this allows the colors to match whatever high-contrast theme the user may have, even a custom one. It allows allows us to remove almost all of the media queries that specifically targeted black-on-white mode.

Additionally, in the process of making these changes I did an audit of all the components in high-contrast mode for both Edge and IE11 and fixed any apparent issues, some of which were already issues in the backlog which are labeled above.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Rating/Rating.scss [hunk 1]
```diff
@@ -41,13 +41,25 @@ $Rating-largeIconSize: 20px;
...
 
+    @media screen and (-ms-high-contrast: active) {
+      color: WindowText;
+    }
+
     &.starIsSelected {
       color: $Rating-selectedColor;
+
+      @media screen and (-ms-high-contrast: active) {
+        color: Highlight;
```
Comment (Jahnp) [inline packages/office-ui-fabric-react/src/components/Rating/Rating.scss]: As a general pattern, I'm wondering if this should be `Highlight` as you have here or `HighlightText`. From skimming [the W3 spec for CSS system colors](https://www.w3.org/TR/css3-color/#css-system), it seems that `Highlight` and `Window` are intended for backgrounds, whereas the -`Text` variants are for foreground. @betrue-final-final--thoughts?

Comment (betrue-final-final) [inline packages/office-ui-fabric-react/src/components/Rating/Rating.scss]: You are correct @jahnp, window and highlight are backgrounds. Windowtext and highlighttext are foregrounds.

Comment (kristoferbrown) [inline packages/office-ui-fabric-react/src/components/Rating/Rating.scss]: @Jahnp @betrue-final-final, the issue with that is that in about half of the default HC palettes `Window` and `HighlightText` are the same color, so `color: HighlightText;` should never be used unless `background: Highlight;` is also set.

There may be a few other places where I've set text color to `Highlight` to indicate some sort of selection or emphasis, and I'll double check if any of those are inappropriate. However, this particular instance in Rating setting color changes the color of the filled in star characters, which is arguably more like a fill color than a text color, so I feel `Highlight` is the correct color in this instance.

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.scss [hunk 2]
```diff
@@ -124,9 +130,17 @@ $detailsList-item-focus-meta-text-color: $ms-color-neutralDark;
     .cell {
       color: $detailsList-item-selected-hover-meta-text-color;
 
+      @media screen and (-ms-high-contrast: active) {
```
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.scss]: can we use the `high-contrast` mixin?

Comment (kristoferbrown) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.scss]: There wasn't a `high-contrast` mixin available outside of fabric-website so I added one to common for now. Seems like there is an open feature request in fabric-core to add these mixins, but for now they're in fabric react common.

**Comment by kristoferbrown** [comment]
@Jahnp, @cschleiden, @atneik, @micahgodbolt, @johannao76, would love a review from one of the code owners on this so we can get this aging PR out the door.

**Comment by betrue-final-final** [comment]
So awesome.

**Comment by micahgodbolt** [comment]
Can you provide some screenshots of affect components? Otherwise, this is cool! Much better to tie to system values than something that might be tweaks by site themes.

**Comment by betrue-final-final** [comment]
I've already been using this method. It works very well.

**Comment by kristoferbrown** [comment]
@micahgodbolt, for resources here's a codepen where you can see how all the different system color names change with your high contrast palette. https://codepen.io/majornista/details/JNjrwx

Most components changed by this will still look the same before and after, but I'll add a few screenshots shortly of components that do change.

**Comment by kristoferbrown** [comment]
@micahgodbolt, there really isn't too much visible to look at in this PR, but here are a few components in high contrast black 

rating
rating
sombobox
sombobox
detailslist
picker


picker

**Comment by kristoferbrown** [comment]
@micahgodbolt, your'e the only remaining code owner approval needed here. Any chance you can take a look here?

**Comment by micahgodbolt** [comment]
apologies :D Wonder if there's a way to filter PRs by the one's that you're blocking

**Pull request merged**